compiler/vmgen.nim

```diff
@@ -453,7 +453,12 @@ proc sameConstant*(a, b: PNode): bool =
     of nkSym: result = a.sym == b.sym
     of nkIdent: result = a.ident.id == b.ident.id
     of nkCharLit..nkUInt64Lit: result = a.intVal == b.intVal
-    of nkFloatLit..nkFloat64Lit: result = a.floatVal == b.floatVal
+    of nkFloatLit..nkFloat64Lit:
+      result = cast[int](a.floatVal) == cast[int](b.floatVal)
```

Comment:
This should be ``cast[int64]`` or ``cast[uint64]``

Reply:
fixed; indeed; hopefully this'll fix i386